mobile/lib/features/text_management/data/repositories/text_repository_impl.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:ffi';
+
+import 'package:flutter/foundation.dart';
+import 'package:mobile/core/error/exceptions.dart';
+import 'package:mobile/core/error/failures.dart';
+import 'package:dartz/dartz.dart';
+import 'package:mobile/features/class_management/data/models/classroom_model.dart';
+import 'package:mobile/features/class_management/domain/entities/classroom.dart';
+import 'package:mobile/features/text_management/data/models/text_model.dart';
+import 'package:mobile/features/text_management/domain/repositories/text_repository.dart';
+import 'package:mobile/features/text_management/domain/entities/text.dart';
+
+import '../data_sources/text_local_data_source.dart';
+
+class TextRepositoryImpl implements TextRepository {
+  final TextLocalDataSource localDataSource;
+
+  TextRepositoryImpl({
+    @required this.localDataSource,
+  });
+
+  @override
+  Future<Either<Failure, Text>> createText(Text text) async {
+    return await _tryCacheText(text);
+  }
+
+  @override
+  Future<Either<Failure, void>> deleteText(Text text) async {
+    return await _tryDeleteText(text);
+  }
+
+  Future<Either<Failure, void>> _tryDeleteText(Text text) async {
+    try {
+      var model = textEntityToModel(text);
+      await localDataSource.deleteTextFromCache(model);
+      return null;
+    } on CacheException {
+      return Left(CacheFailure());
+    }
+  }
+
+  Future<Either<Failure, Text>> _tryCacheText(Text text) async {
+    try {
+      var model = textEntityToModel(text);
+      var localModel = await localDataSource.cacheNewText(model);
+      var localText = textModelToEntity(localModel);
+      return Right(localText);
+    } on CacheException {
+      return Left(CacheFailure());
+    }
+  }
+
+  @override
+  Future<Either<Failure, List<Text>>> getTexts(Classroom classroom) async {
```

Comment:
Just a short note:

It's good practice to have your interfaces (public methods) higher in the class, before the private methods